provider/provider.go

```diff
@@ -236,7 +243,9 @@ func (p *Provider) fetchAccount(accountID string, awsCfg aws.Config, svc *sts.Cl
 
 		innerLog := p.Logger.With("account_id", accountID, "region", region)
 		for serviceName, newFunc := range regionalServices {
-			resourceClients[serviceName] = newFunc(awsCfg,
+			cfg := awsCfg.Copy()
```

Comment:
I think that there is no need for this line as it is already copied when `awsCfg` pass to this function (as it's not a pointer).